### ISSUE TITLE
Extending Adaptive Methods for Finding an Optimal Circuit Ansatze in VQE Optimization

### ISSUE THREAD

**Issue by camponogaraviera** [XanaduAI/QHack2022#65]
# Title

Extending Adaptive Methods for Finding an Optimal Circuit Ansatze in VQE Optimization

# Team Name:

Hackeinberg.

# Project Description:

Most widely considered hardware-efficient and Chemistry-inspired ansatze, although generic, suffer from either barren plateaus [[1](https://www.nature.com/articles/s41467-018-07090-4)] or inconsistency under low-order trotterization steps [[2](https://pubs.acs.org/doi/abs/10.1021/acs.jctc.9b01083)], respectively. To circumvent this drawback, adaptive circuits have already been implemented in the literature [[3](https://www.nature.com/articles/s41467-019-10988-2)] [[4](https://pennylane.ai/qml/demos/tutorial_adaptive_circuits.html)]. We propose to extend the existing methods applied to the hybrid quantum-classical VQE [[5](https://doi.org/10.1038/ncomms5213)] algorithm under a case study on the ground state of the LiH molecule. To this end, the Qamuy SDK will be leveraged with seamless integration with another framework of choice (PennyLane, Qiskit, Cirq...). Here, we propose and demonstrate an approach to find the best gate arrangement for a circuit ansatz according to an optimization method satisfying the following constraints/features for a good circuit Ansatz:

1. Coherence friendly: the circuit must be shallow, i.e, have a small number of layers in order to be computed during a time window smaller than the decoherence time.
2. Hardware friendly (qubit routing): gate coupling allowed only between nearest-neighbor qubits to avoid SWAP gates during qubit routing (mapping from the circuit diagram to a hardware topology).
3. Small number of hyperparameters: we seek the minimum amount of angles to be optimized in order to avoid classical optimization overhead (when classical computation becomes too expensive).

Finally, we benchmark our results against the standard UCCSD ansatz approach.

# Source code:

[GitHub](https://github.com/QuCAI-Lab/qhack2022-hackeinberg-project/blob/dev/README.md).

# Resource Estimate:

We leverage the [Qamuy SDK](https://qamuy.qunasys.com/docs/en/) cloud computing.

# Challenges:

- [Quantum Chemistry Challenge](https://github.com/XanaduAI/QHack/blob/master/Open_Hackathon.md#quantum-chemistry-challenge).
- [Science Challenge](https://github.com/XanaduAI/QHack/blob/master/Open_Hackathon.md#science-challenge).

# Team Members:

Developed by [@camponogaraviera](https://github.com/camponogaraviera) and [@zemarchezi](https://github.com/zemarchezi).

# References

\[1] McClean, J.R., Boixo, S., Smelyanskiy, V.N. et al. Barren plateaus in quantum neural network training landscapes. [Nat Commun 9, 4812 (2018)](https://www.nature.com/articles/s41467-018-07090-4). 

\[2] Grimsley, H. R.; Claudino, D.; Economou, S. E.; Barnes, E.; Mayhall, N. J. Is the trotterized uccsd ansatz chemically well-defined? [J. Chem. Theory Comput. 2020, 16, 1](https://pubs.acs.org/doi/abs/10.1021/acs.jctc.9b01083).

[3] Harper R. Grimsley, Sophia E. Economou, Edwin Barnes, Nicholas J. Mayhall, “An adaptive variational algorithm for exact molecular simulations on a quantum computer”. [Nat. Commun. 2019, 10, 3007](https://www.nature.com/articles/s41467-019-10988-2).

[4] PennyLane dev team, "Adaptive circuits for quantum chemistry". [PennyLane, 13 September 2021](https://pennylane.ai/qml/demos/tutorial_adaptive_circuits.html).

[5] Peruzzo, A., McClean, J., Shadbolt, P. et al. A variational eigenvalue solver on a photonic quantum processor. [Nat Commun 5, 4213 (2014)](https://doi.org/10.1038/ncomms5213).

**Comment by isaacdevlugt** [comment]
Thank you for your Power Up submission! As a reminder, the final deadline for your project is February 25 at 17h00 EST. Submissions should be done here: https://github.com/XanaduAI/QHack/issues/new?assignees=&labels=&template=open_hackathon.md&title=%5BENTRY%5D+Your+Project+Title

This issue will be closed shortly.

Good luck!

**Comment by isaacdevlugt** [comment]
Thank you for your Power Up submission! As a reminder, the final deadline for your project is February 25 at 17h00 EST. Submissions should be done here: https://github.com/XanaduAI/QHack/issues/new?assignees=&labels=&template=open_hackathon.md&title=%5BENTRY%5D+Your+Project+Title

This issue will be closed shortly.

Good luck!
